back/src/commands/onboarding.helpers.ts

```diff
@@ -216,3 +219,84 @@ export const sendPendingMembershipRequestDetailsEmail = async () => {
 
   await prisma.$disconnect();
 };
+
+export const getPendingMembershipRequestsAndAssociatedAdmins = async (
+  daysAgo: number
+) => {
+  const now = new Date();
+
+  const requestDateGt = xDaysAgo(now, daysAgo);
+  const requestDateLt = xDaysAgo(now, daysAgo - 1);
+
+  // First, fetch all unanswered membership requests
+  const pendingMembershipRequests = await prisma.membershipRequest.findMany({
+    where: {
+      createdAt: { gte: requestDateGt, lt: requestDateLt },
+      status: MembershipRequestStatus.PENDING
+    },
+    include: {
+      // We need the issuer email and company orgId in the email
+      user: { select: { email: true, isActive: true } },
+      company: { select: { name: true, orgId: true } }
+    }
+  });
+
+  // Keep only active users
+  const activePendingMembershipRequests = pendingMembershipRequests.filter(
+    r => r.user.isActive
+  );
+
+  // Get unique companyIds
+  const companyIds = [
+    ...new Set(activePendingMembershipRequests.map(p => p.companyId))
+  ];
+
+  // Get all the admins from all those companies
+  const admins = await getActiveAdminsByCompanyIds(companyIds);
+
+  return activePendingMembershipRequests.map(request => {
+    const requestAdmins = admins.filter(a => a.companyId === request.companyId);
+
+    return { ...request, admins: requestAdmins };
+  });
+};
+
+/**
+ * For each unanswered membership request issued X days ago, send an
+ * email to all the admins of request's targeted company
+ */
+export const sendPendingMembershipRequestToAdminDetailsEmail = async () => {
+  const requests = await getPendingMembershipRequestsAndAssociatedAdmins(14);
+
+  const messageVersions: MessageVersion[] = requests.map(request => {
+    const variables = {
+      requestId: request.id,
+      email: request.user.email,
+      orgName: request.company.name,
+      orgId: request.company.orgId
+    };
+
+    const template = renderMail(pendingMembershipRequestAdminDetailsEmail, {
```

Comment:
Bémol ici. La façon dont on a construit `renderMail` est un peu bancale, parce qu'en fait cette méthode, qui prend des paramètres et un template, ne se contente pas de retourner le template populé, mais le mail en entier (avec `to`, `cc`, `urls` etc.)

Du coup ça pose un peu problème parce que pour les `messageVersions` je voudrais juste le template populé. J'ai fait ce "hack" qui est de quand même utiliser cette fonction et de juste ignorer tout ce dont je n'ai pas besoin, mais j'imagine qu'on pourrait faire ça plus clean et split les fonctions

Reply:
On pourrait yes, en attendant mieux c'est bien comme ça👍